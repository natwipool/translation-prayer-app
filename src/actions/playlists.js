import uuid from 'uuid';

// ADD_PLAYLIST
export const addPlaylist = ({ name = '', lists = [] } = {}) => ({
  type: 'ADD_PLAYLIST',
  playlist: {
    id: uuid(),
    name,
    lists
  }
});

// REMOVE_PLAYLIST
export const removePlaylist = ({ id } = {}) => ({
  type: 'REMOVE_PLAYLIST',
  id
});

// EDIT_PLAYLIST
export const editPlaylist = (id, updates) => ({
  type: 'EDIT_PLAYLIST',
  id,
  updates
});
