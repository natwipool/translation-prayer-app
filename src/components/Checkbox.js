import React from 'react';

// class Checkbox extends React.Component {
//   constructor(props) {
//     super(props)

//     this.state = {
//       checked: props.lists.find((list) => list === props.precept)
//     }
//   }

//   onAddLists = (list) => {
//     this.setState(() => ({
//       checked: !this.state.checked
//     }))
//     console.log(this.state.checked)
//     this.props.onAddLists(list)
//   }
  
//   onRemoveLists = (list) => {
//     this.setState(() => ({
//       checked: !this.state.checked
//     }))
//     this.props.onRemoveLists(list); 
//   }

//   render() {
//     return (
//       <div>
//         <input
//           type="checkbox"
//           name={this.props.precept}
//           value={this.props.precept}
//           checked={this.state.checked}
//           onChange={e => {
//             const list = e.target.value;
//             if (e.target.checked) {
//               this.onAddLists(list)  
//             } else if (!e.target.checked) {
//               this.onRemoveLists(list);  
//             }
//           }}
//         />
//         <label htmlFor={this.props.precept}>
//           {this.props.precept}
//           {
//             this.props.lists.indexOf(this.props.precept) > -1 && (
//               this.props.lists.indexOf(this.props.precept) + 1
//             )
//           }
//         </label>
//       </div>
//     )
//   }
// }

const Checkbox = (props) => (
  <div>
    <input
      type="checkbox"
      name={props.precept}
      value={props.precept}
      checked={props.lists.find((list) => list === props.precept)}
      onChange={e => {
        const list = e.target.value;
        if (e.target.checked) {
          props.onAddLists(list)  
        } else if (!e.target.checked) {
          props.onRemoveLists(list);  
        }
      }}
    />
    <label htmlFor={props.precept}>
      {props.precept}
      {
        props.lists.indexOf(props.precept) > -1 && (
          props.lists.indexOf(props.precept) + 1
        )
      }
    </label>
  </div>
);

export default Checkbox;
