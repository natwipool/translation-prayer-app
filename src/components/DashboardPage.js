import React from 'react';
import PrefacePage from './PrefacePage'

const DashboardPage = () => (
  <div>
    <PrefacePage />
  </div>
);

export default DashboardPage;