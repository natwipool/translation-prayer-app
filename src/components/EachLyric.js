import React from 'react';

const EachLyric = (props) => (
  <div>
    {props.lyric}
  </div>
);

export default EachLyric;