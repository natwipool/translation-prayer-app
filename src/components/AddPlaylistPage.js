import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { addPlaylist } from '../actions/playlists';

export class AddPlaylistPage extends React.Component {
  addPlaylist = playlist => {
    this.props.addPlaylist(playlist);
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <div>
        <h1>เพิ่มรายการสวดมนต์</h1>
        <PlaylistForm onSubmit={this.addPlaylist} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addPlaylist: playlist => dispatch(addPlaylist(playlist))
});

export default connect(undefined, mapDispatchToProps)(AddPlaylistPage);
