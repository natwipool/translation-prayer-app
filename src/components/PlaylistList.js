import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { removePlaylist } from '../actions/playlists';

export const PlaylistList = ({ id, name }) => (
  <div>
    <Link to={`/playlist/${id}`}>
      <h3>{name}</h3>
    </Link> 
  </div>
);

export default connect()(PlaylistList);