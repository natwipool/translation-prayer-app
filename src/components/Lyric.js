import React from 'react';
import EachLyric from './EachLyric';

const Lyric = (props) => {
  return (
    <div>
      <h3>{props.transPrayer.precept}</h3>
      {props.transPrayer.lyrics.map((lyric, index) => 
        <EachLyric 
          key={index}
          lyric={lyric}
        />
      )}
    </div>
  )
}

export default Lyric;