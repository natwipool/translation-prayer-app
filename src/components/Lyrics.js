import React from 'react';
import Lyric from './Lyric';

const Lyrics = (props) => (
  <div>
    {
      props.transPrayers.length === 0 ? (
        <p>ไม่มีบทสวด</p>
      ) : (
        props.transPrayers.filter((transPrayer, index) => 
          index  === props.index
        ).map((result, index) => 
          <Lyric
            key={index} 
            transPrayer={result}
          />
        )
      )
    }
  </div>
)

export default Lyrics;