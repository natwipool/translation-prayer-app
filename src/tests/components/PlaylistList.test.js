import React from 'react';
import { shallow } from 'enzyme';
import { PlaylistList } from '../../components/PlaylistList';
import playlists from '../fixtures/playlists';

test('should render PlaylistList correctly', () => {
  const wrapper = shallow(<PlaylistList {...playlists[1]} />);
  expect(wrapper).toMatchSnapshot();
});