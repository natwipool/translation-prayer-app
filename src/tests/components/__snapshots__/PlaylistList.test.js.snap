// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlaylistList correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/playlist/2"
  >
    <h3>
      Playlist name two
    </h3>
  </Link>
</div>
`;
